### PR TITLE
fix(graph): modernise queries (#5878)

### PR DESCRIPTION
**Pull request by jeremyjaouen**
## Description

Backport of #5878

**Fixes** # MON-154762

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 22.10.x
- [ ] 23.04.x
- [x] 23.10.x
- [ ] 24.04.x
- [ ] 24.10.x
- [ ] master

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: centreon/www/class/centreonGraph.class.php [hunk 1]
```diff
@@ -1760,12 +1800,12 @@ private function manageVMetric($vId, $vName, $indexId)
...
                         }
-                    } elseif ($l_poqy->rowCount() == 0) {
+                    } elseif ($checkRealMetricStatement->rowCount() === 0) {
                         /* key : id or vname and iid */
-                        $l_poqy->closeCursor();
+                        $this->DB->closeQuery($checkRealMetricStatement);
                         $this->manageVMetric(null, $l_mnane, $lVmetric["index_id"]);
                     } else {
-                        $l_poqy->closeCursor();
+                        $this->DB->closeQuery($checkRealMetricStatement);
```
Comment (Drago37) [inline centreon/www/class/centreonGraph.class.php:1808]: Normal to close the connection during the loop ? You can close after, line 1811for example.

File: centreon/www/class/centreonGraph.class.php [hunk 2]
```diff
@@ -271,13 +271,20 @@ public function __construct($user_id, $index = null, $debug = 0, $compress = nul
...
+            $insertViewDetailsStatement = $this->DB->prepareQuery(
+                <<<SQL
+                    INSERT INTO `ods_view_details`
+                    (`metric_id`, `contact_id`, `all_user`, `index_id`)
+                    VALUES (:metric_id, :user_id, '0', :index_id)
+                    SQL
+            );
+            foreach (array_keys($odsm) as $mid) {
+                if (! isset($metrics_cache[$mid])) {
+                    $this->DB->executePreparedQuery($insertViewDetailsStatement, [
```
Comment (Drago37) [inline centreon/www/class/centreonGraph.class.php:283]: Catch CentreonDbException and log the error with CentreonLog with business informations and a context please

**Review by Drago37 — CHANGES_REQUESTED** [review]
To fix for all the file at several locations thanks

File: centreon/www/class/centreonGraph.class.php (hunk 1 above)
Comment (jeremyjaouen) [reply centreon/www/class/centreonGraph.class.php:1808]: I don't want to change the existing behaviour

Comment (Drago37) [reply centreon/www/class/centreonGraph.class.php:1808]: ok

File: centreon/www/class/centreonGraph.class.php (hunk 2 above)
Comment (Drago37) [reply centreon/www/class/centreonGraph.class.php:283]: Seen together, you don't want to change your code this time to merge it quickly. 
But the next time, can you catch the exception and use the logger please to improve our error handling ? Thanks.

**Review by Drago37 — APPROVED** [review]

**Comment by jeremyjaouen** [comment]
will be sync after release

**Pull request closed without merging**